.github/DESIGN.md

```diff
@@ -0,0 +1,115 @@
+To help visualize all the components and states avialable in the Shopify App CLI, we have crteated a [UI Kit](https://www.figma.com/file/ZXIgM4wQpfRNjGaIArjWOgTD/CLI-UI-Kit?node-id=67%3A0) in [Figma](http://figma.com) that you can use to build command flows.
+
+*Figma is a free web based design tool.*
+
+## Assumptions
+The user understands the basic mechanics of a cli:
+- can type in commands to execute tasks 
+- there is a persistent “help” command that will educate them on specifics of each command/subcommand
+- CNTRL + C quits any running task
+- how to navigate and manipulate the filesystem via the command line: cd and mkdir
+
+## Design principles
+High level guiding principles to help make decisions faster when creating a CLI UI.
+
+**Communicate every state (success, waiting, error) even if it’s redundant.**  
+
+✅ Communicate task success even if the external process has also communicated success. Always wrap up what just happened with a success/error message.  
+❌ Assume that success is understood because the process finished. 
+
+**Suppress output when the process is successful, show output when the process has an error.**  
+
+✅ Hide all STDOUT information if the process has run successfully, show all output if the process failed.  
+❌ Show every line or state if the process success/error state has not been determined yet.
+
+**Error out if a task needs to make a system change and tell the user how to complete the installation.**  
+
+✅ Prompt the user if you need to run something external that will modify their system to complete the installation.  
+❌ Automatically install an NPM package that modifies the system. Prompt them that it’s needed and ask them if they would like the cli to install it.
+
+**If a process that fails early will affect the rest of the task, terminate it as soon as possible and tell the user why.**  
+
+✅ Error out of a process if you know the rest of the task will fail.  
+❌ Continue to try successfully completing the task if one crucial task failed.
+
+**Output one idea per line.**  
+
+✅ Error: you are not in an app project folder  
+✅ Tip: use shopify create project to create a new app project  
+❌ Error: you are not in an app project folder, use shopify create project to create a new app project.
+
+**The command should ask for all the information needed to execute the task automatically.**  
+
+✅ Have opinion on smart defaults that the user can override with arguments and options.  
+❌ Defer multiple options or inputs to after the command is executing.
+
+**Let the user opt in to verbosity.**  
+
+✅ Show the user whatever is necessary given the task at hand. A summary may be best suited for tasks with many sub processes, where some output might make sense for small tasks.  
+❌ Show the user every bit of information that comes in from external or internal tasks.
+
+## CLI UI states
+CLI’s tend to have less states that GUI, however error states are more prevalent because a CLI can error out at any moment.
```

Comment:
Is this sentence true?

Reply:
should be "CLIs have fewer states than GUIs"
- no apostrophe in CLIs
- than vs. that

Perhaps instead something that doesn't get into "prevalance" of errors, which I don't think is meaningfully different in CLIs vs GUIs:

> Because of the serial nature of terminal input and output, CLIs have fewer states than graphical user interfaces. We provide design guidelines for three states: